src/proto/connection.rs

```diff
@@ -56,13 +56,14 @@ impl<T, P, B> Connection<T, P, B>
           P: Peer,
           B: IntoBuf,
 {
-    pub fn new(codec: Codec<T, Prioritized<B::Buf>>) -> Connection<T, P, B> {
+    pub fn new(codec: Codec<T, Prioritized<B::Buf>>, settings: frame::Settings) -> Connection<T, P, B> {
```

Comment:
This could probably be passed in by ref and it would avoid the clone / move from above.